web-local/src/lib/application-state-stores/common-store.ts

```diff
@@ -0,0 +1,77 @@
+import type { V1ReconcileError } from "@rilldata/web-common/runtime-client";
+import { Readable, writable } from "svelte/store";
+
+export type CommonEntityData = {
+  errors: Array<V1ReconcileError>;
+};
+
+/**
+ * Store to save common data across entities.
+ * Currently only has errors
+ */
+export type CommonEntityState = {
+  entities: Record<string, CommonEntityData>;
+};
+const { update, subscribe } = writable({
+  entities: {},
+} as CommonEntityState);
+
+const createOrUpdateEntity = (
+  path: string,
+  callback: (entityData: CommonEntityData) => void
+) => {
+  update((state) => {
+    if (!state[path]) {
+      state.entities[path] = {
+        errors: [],
+      };
+    }
+    callback(state.entities[path]);
+    return state;
+  });
+};
+
+const commonEntitiesReducers = {
+  setErrors(path: string, errors: Array<V1ReconcileError>) {
+    createOrUpdateEntity(path, (entityData: CommonEntityData) => {
+      entityData.errors = errors;
+    });
+  },
+
+  consolidateMigrateResponse(
```

Comment:
This function name is confusing too. What if we called this `setErrors`, since that's more readable at the component level, and removed the current `setErrors` function, which is only called once?